src/Models/NonhydrostaticModels/conjugate_gradient_poisson_solver.jl

```diff
@@ -133,6 +158,12 @@ Base.summary(::DiagonallyDominantPreconditioner) = "DiagonallyDominantPreconditi
     arch = architecture(p)
     fill_halo_regions!(r)
     launch!(arch, grid, :xyz, _diagonally_dominant_precondition!, p, grid, r)
+
+    P = mean(p)
+    grid = solver.grid
+    arch = architecture(grid)
+    launch!(arch, grid, :xyz, subtract_and_mask!, p, grid, P)
```

Comment:
@xkykai I think this is the crucial lines. Without this the pressure does _not_ have a zero mean and I suspect that can throw off the CG solver. But I'm checking.

Reply:
@jm-c might be good to have your input

Reply:
I have found that I needed to enforce the zero mean when I was using the conjugate gradient solver with @elise-palethorpe to compare with MultiGrid

Reply:
I think this was a key ingredient missing from previous implementation. The FFT-based preconditioner zeros out the mean over the underlying grid, but does not zero out the mean on the immersed grid.

Reply:
Checked the derivation, looks good to me!